components/director/internal/domain/formationassignment/status_service.go

```diff
@@ -116,11 +116,13 @@ func (fau *formationAssignmentStatusService) DeleteWithConstraints(ctx context.C
 		return errors.Wrapf(err, "while enforcing constraints for target operation %q and constraint type %q", model.NotificationStatusReturned, model.PreOperation)
 	}
 
-	if err = fau.repo.Delete(ctx, id, tenantID); err != nil {
-		if apperrors.IsUnauthorizedError(err) {
-			return apperrors.NewNotFoundError(resource.FormationAssignment, id)
+	if notificationStatusReport.State == string(model.ReadyAssignmentState) {
+		if err = fau.repo.Delete(ctx, id, tenantID); err != nil {
+			if apperrors.IsUnauthorizedError(err) {
```

Comment:
Why for unauthoriazed error we return a NotFound Err? Probably it is intended, but I'm curious about it

Reply:
This is due to the way the repo layer `Update` and `Delete` methods work, in cases when there are 0 rows affected in the SQL statement, they return `Unauthorized` rather than `NotFound`. This combined with the transaction isolation level, if some other transaction deletes the assignment before this one, it would return `Unauthorized`, even though this transaction could get the entity a few lines above in the same transaction.